pftools/python/parflow/tools/builders.py

```diff
@@ -603,6 +607,14 @@ def apply(self, run=None, name_registration=True):
         else:
             self.run = run
 
+        # new names are user specified names such as s1 s2 s3 for soil permeability
+        if infer_key_names and name_registration:
+            new_names = ""
+            for name in self.output:
+                if isinstance(self.output[name], dict):
+                    new_names = new_names + name + " "
+            self.key_root.Names = new_names
+
```

Comment:
The above lines support putting the names in the table and not needing to declare them separately in the run file

Reply:
I've added a flag infer_key_names just in case there are unintended consequences I didn't catch. Flag is default set to false so this shouldn't break backwards compatibility